presto-main/src/test/java/com/facebook/presto/operator/scalar/AbstractTestFunctions.java

```diff
@@ -42,45 +44,61 @@ protected void assertInvalidFunction(String projection, Type expectedType, Strin
     {
         try {
             assertFunction(projection, expectedType, null);
-            fail("Expected to throw an INVALID_FUNCTION_ARGUMENT exception with message " + message);
         }
         catch (PrestoException e) {
             assertEquals(e.getErrorCode(), INVALID_FUNCTION_ARGUMENT.toErrorCode());
             assertEquals(e.getMessage(), message);
+            return;
         }
+        fail("Expected to throw an INVALID_FUNCTION_ARGUMENT exception with message " + message);
```

Comment:
Because you asked me to :)

https://github.com/facebook/presto/pull/3710#discussion-diff-42404960